### PR TITLE
Updated and added unit tests

### PR DESCRIPTION
**Pull request by edking21**
Tested valid and invalid email options.

### CODE REVIEW

File: config.json.example [hunk 1]
```diff
@@ -15,7 +15,14 @@
...
     ],
-
+    "ccAllowList": [
+      "user@localdev",
+      "test@localdev"
+    ],
+    "bccAllowList": [
+      "user@localdev",
+      "test@localdev"
+    ],
```
Comment (damianb) [inline config.json.example:25]: There is no ccAllowList or bccAllowList functionality defined.  This is also not something that we have a use-case for implementation either, similar to the prior discussion on toAllowList.

Only senderAllowList and replyToAllowList are defined.  Reference: https://github.com/uncomnllc/gfm-mailer-prototype/blob/master/src/interfaces/AppConfig.ts#L36-L37

File: src/server.ts [hunk 2]
```diff
@@ -256,9 +256,9 @@ export class NotifyServer {
             throw new Error('Specified sender email is not in sender allowlist.')
           }
 
-          if (payload.replyTo && !this.config.api.replyToAllowList.includes(payload.replyTo)) {
-            throw new Error('Specified reply-to email is not in replyto allowlist.')
-          }
+          // if (payload.replyTo && !this.config.api.replyToAllowList.includes(payload.replyTo)) {
+          //   throw new Error('Specified reply-to email is not in replyto allowlist.')
+          // }
```
Comment (damianb) [inline src/server.ts:261]: Why was this commented out?

File: tests/server.spec.ts [hunk 3]
```diff
@@ -80,8 +80,145 @@ describe('notification server', () => {
...
+    const res = await chai.request(`https://localhost:${port.api}`)
+      .post('/dispatch')
+      .ca(creds.notifyCert)
+      .cert(creds.goodClientCert)
+      .key(creds.goodClientKey)
+      .send({
+        sender: 'sender@localhost',
+        to: ['test@localhost'],
+        subject: 'test email',
+        replyTo: ['bogus@localhost'],
```
Comment (damianb) [inline tests/server.spec.ts:116]: replyTo should be of datatype `string`, not `string[]`

File: tests/server.spec.ts [hunk 4]
```diff
@@ -80,8 +80,145 @@ describe('notification server', () => {
...
+    it('should reject if a cc is not in the allow list', async () => {
+      const res = await chai.request(`https://localhost:${port.api}`)
+        .post('/dispatch')
+        .ca(creds.notifyCert)
+        .cert(creds.goodClientCert)
+        .key(creds.goodClientKey)
+        .send({
+          sender: 'sender@localhost',
+          to: ['test@localhost'],
+          cc: 'bogus@localhost',
```
Comment (damianb) [inline tests/server.spec.ts:159]: cc should be of type `string[]` and not `string`.

also, per above comment, cc is not something we are implementing an allowList for.

File: src/interfaces/AppConfig.ts [hunk 5]
```diff
@@ -34,7 +34,9 @@ export const AppConfig = Record({
     authorizedCertsPath: String,
 
     senderAllowList: Array(String),
-    replyToAllowList: Array(String)
+    replyToAllowList: String,
```
Comment (damianb) [inline src/interfaces/AppConfig.ts:37]: replyToAllowList should remain as the type `string[]`.  We need to be able to allow multiple replyTo addresses that are allowed to be specified, but only permit specifying one such address during the API call.

File: config.json.example (hunk 1 above)
Comment (edking21) [reply config.json.example:25]: All code to check the allow list for cc and bcc has been removed.

File: src/server.ts (hunk 2 above)
Comment (edking21) [reply src/server.ts:261]: My bad.  This was commented out to fix a compile problem  when I not checking the AllowList for replyTo.  Now we are checking the allowList for replyTo so the code has been re-added.

File: tests/server.spec.ts (hunk 3 above)
Comment (edking21) [reply tests/server.spec.ts:116]: I have changed replyTo to datatype string.  Test is passing.

File: tests/server.spec.ts (hunk 4 above)
Comment (edking21) [reply tests/server.spec.ts:159]: The test you are showing here has been removed so there is not change from string to string[].  The test was checking an allow list for cc.

File: src/interfaces/AppConfig.ts (hunk 5 above)
Comment (edking21) [reply src/interfaces/AppConfig.ts:37]: replyToAllowList in AppConfig.ts has been reverted back to be an array.  The test is passing.

**Comment by edking21** [comment]
Validating a  to, cc and bcc could be done with a regular expression.  There are several examples on the web.  Writing the code to confirm the address passes the regex test is easy.  Integrating into the current code will be more difficult.  I will need to setup a screen share to get your help on integrating the to, cc and bcc check into the code.

File: tests/nodemailer-stub.d.ts [hunk 6]
```diff
@@ -14,6 +14,9 @@ declare module 'nodemailer-stub' {
     envelope: MimeNode.Envelope
     from: string
     to: string
+    cc: string[]
+    bcc: string[]
+    replyTo: string
```
Comment (damianb) [inline tests/nodemailer-stub.d.ts:19]: These properties do not exist on the IWMMail object and should not be added.  Please remove.

File: src/interfaces/AppConfig.ts [hunk 7]
```diff
@@ -34,7 +34,7 @@ export const AppConfig = Record({
     authorizedCertsPath: String,
 
     senderAllowList: Array(String),
-    replyToAllowList: Array(String)
+    replyToAllowList: Array(String),
```
Comment (damianb) [inline src/interfaces/AppConfig.ts:37]: This does not pass eslint.  Please remediate.

File: src/server.ts [hunk 8]
```diff
@@ -255,10 +255,11 @@ export class NotifyServer {
           if (!this.config.api.senderAllowList.includes(payload.sender)) {
             throw new Error('Specified sender email is not in sender allowlist.')
           }
-
+          
           if (payload.replyTo && !this.config.api.replyToAllowList.includes(payload.replyTo)) {
             throw new Error('Specified reply-to email is not in replyto allowlist.')
           }
+
```
Comment (damianb) [inline src/server.ts:262]: This does not pass eslint.  Please remediate.

**Pull request merged**